src/Filesystem/Wrappers/File.php

```diff
@@ -3,19 +3,72 @@
 namespace StrictPhp\HttpClients\Filesystem\Wrappers;
 
 use SplFileObject;
-use StrictPhp\HttpClients\Filesystem\Contracts\FileContract;
+use StrictPhp\HttpClients\Filesystem\Contracts\FileInterface;
 
-final class File implements FileContract
+final class File implements FileInterface
 {
-    private readonly SplFileObject $file;
+    private ?SplFileObject $file = null;
 
-    public function __construct(string $path)
-    {
-        $this->file = new SplFileObject($path, 'w');
+    public function __construct(
+        private readonly string $pathname,
+    ) {
     }
 
     public function write(string $content): void
     {
-        $this->file->fwrite($content);
+        $this->getFile()
+            ->fwrite($content);
+    }
+
+    public function content(): ?string
+    {
+        $file = $this->readFile();
+        if (! $file instanceof SplFileObject) {
+            return null;
+        } elseif ($file->getMTime() !== $file->getCTime() && $file->getMTime() < time()) {
+            $this->remove();
+            return null;
+        }
+
+        $content = $file->fread($file->getSize());
+        if ($content === false) {
+            return null;
+        }
+
+        return $content;
+    }
+
+    public function remove(): void
+    {
+        $file = $this->readFile();
+        if ($file instanceof SplFileObject) {
+            unlink($file->getPathname());
+            $this->file = null;
+        }
+    }
+
+    public function setTtl(?int $ttlToSeconds): void
+    {
+        /** @var int $ctime - file exists! */
+        $ctime = $this->getFile()
+            ->getCTime();
+        touch($this->pathname, $ctime + (int) $ttlToSeconds);
```

Comment:
implementoval jsem ttl pro keš, tak že mám ctime, pokud se mi rovná s mtime, tak je to bez omezení, pokud se nerovná, tak platí expirace mtime.